pkg/gohai/filesystem/filesystem_nix.go

```diff
@@ -0,0 +1,196 @@
+// This file is licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+//go:build darwin || linux
+// +build darwin linux
```

Comment:
Note that the minimal supported version of Go for Gohai is 1.18 since it is used outside of the agent, but as Go 1.18 already supported the new build constraints I'll remove it.
I'll open a follow-up PR to clean that in the whole package.

Reply:
Done in d39af360e3e37d007a32be76448b92c0c52753da